Neo4j.Driver/Neo4j.Driver.Tests/Routing/LoadBalancerTests.cs

```diff
@@ -0,0 +1,561 @@
+                        throw balancer.CreateClusterPooledConnectionErrorHandler(uri)
+                            .OnConnectionError(new AuthenticationException("Failed to auth the client to the server."));
+                    });
+                    
+                    // When
+                    var error = Record.Exception(() => balancer.UpdateRoutingTable());
+
+                    // Then
+                    error.Should().BeOfType<AuthenticationException>();
+                    error.Message.Should().Contain("Failed to auth the client to the server.");
+
+                    // while the server is not removed
+                    routingTable.All().Should().ContainInOrder(uri);
+                }
+            }
+
+            public class AcquireReadWriteConnectionMethod
+            {
+                private static IRoutingTable CreateRoutingTable(Uri uri, AccessMode mode)
+                {
+                    IRoutingTable routingTable;
+                    switch (mode)
+                    {
+                        case AccessMode.Read:
+                            routingTable = NewRoutingTable(readers: new[] {uri});
+                            break;
+                        case AccessMode.Write:
+                            routingTable = NewRoutingTable(writers: new[] {uri});
+                            break;
+                        default:
+                            throw new InvalidOperationException($"Unknown type {mode} to this test.");
+                    }
+                    return routingTable;
+                }
+
+                private static IPooledConnection AcquiredConn(LoadBalancer balancer, AccessMode mode)
+                {
+                    IPooledConnection acquiredConn;
+                    switch (mode)
+                    {
+                        case AccessMode.Read:
+                            acquiredConn = balancer.AcquireReadConnection();
+                            break;
+                        case AccessMode.Write:
+                            acquiredConn = balancer.AcquireWriteConnection();
+                            break;
+                        default:
+                            throw new InvalidOperationException($"Unknown type {mode} to this test.");
+                    }
+                    return acquiredConn;
+                }
+
+                [Theory]
+                [InlineData(AccessMode.Read)]
+                [InlineData(AccessMode.Write)]
+                public void ShouldThrowSessionExpiredExceptionIfNoServerAvailable(AccessMode mode)
+                {
+                    // Given
+                    var routingTable = NewRoutingTable();
+                    var balancer = new LoadBalancer(null, routingTable);
+
+                    // When
+                    var error = Record.Exception(()=>AcquiredConn(balancer, mode));
+
+                    // Then
+                    error.Should().BeOfType<SessionExpiredException>();
```

Comment:
This class is to test the acquireReadConn or acquireWriteConn methods without ensureRouting. That is to say if we already assume that the routing table we obtained is a valid one, then test what conn we will obtain from acquireRead/WriteConn method call. 

This method tests that if the routing table is valid but we failed to connect to any of the server in routing table, then we should session expired and let user retry.

If we consider ensureRouting, then this should definitely be a SU.